File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        maxWater = 0
        l, r = 0, len(heights) - 1

        while l < r:
            d = r - l

            if heights[l] > heights[r]:
                w = d * heights[r]
                maxWater = max(maxWater, w)
                r -= 1
            else:
                w = d * heights[l]
                maxWater = max(maxWater, w)
                l += 1
        
        return maxWater

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.js
class Solution {
    maxProfit(prices) {
        let maxProfit = 0;
        let buy = prices[0];

        for (let i = 1; i < prices.length; i++) {
            buy = Math.min(buy, prices[i - 1]);
            const sell = prices[i];
            const profit = sell - buy;
            maxProfit = Math.max(maxProfit, profit);
        }

        return maxProfit;
    }
}
